### PR TITLE
Fix/GRW-1108/CPAS BackButton fix

### PR DESCRIPTION
**Pull request by marian-buc**
## What?
Added direct links to checkout pages BackButton, instead of defaulting to history.goBack()

## Why?
To prevent redirecting outside of the app when users arrive by entering the link manually

**Ticket(s): [GRW-1108]**

[GRW-1108]: https://hedvig.atlassian.net/browse/GRW-1108?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/client/pages/Checkout/CheckoutPayment/CheckoutPayment.tsx [hunk 1]
```diff
@@ -430,7 +429,9 @@ export const CheckoutPayment = ({
 
   const handleClickBackButton = () => {
     trackOfferEvent({ eventName: EventName.ContactInformationPageGoBack })
+    history.push(detailsPageLink)
   }
+  const detailsPageLink = `/${locale.path}/new-member/checkout/details/${quoteCartId}`
```
Comment (robinandeer) [inline src/client/pages/Checkout/CheckoutPayment/CheckoutPayment.tsx:434]: Let's move this inside the `handleClickBackButton` function. Looks odd that it's defined after the function :/

File: src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx [hunk 2]
```diff
@@ -74,6 +78,7 @@ export const CheckoutDetails = () => {
   const quoteDetails = data.quoteDetails
   const bundleVariants = data.bundleVariants
   const allQuotes = getUniqueQuotesFromVariantList(bundleVariants)
+  const offerPageLink = `/${localePath}/new-member/offer/${quoteCartId}`
```
Comment (robinandeer) [inline src/client/pages/Checkout/CheckoutDetails/CheckoutDetails.tsx:81]: Hmm also here, any reason to define it outside the function that uses this variable?

File: src/client/components/BackButton/BackButton.tsx [hunk 3]
```diff
@@ -24,9 +24,8 @@ export const BackButton = ({ onClick }: Props) => {
   const history = useHistory()
 
   const handleClick = () => {
-    history.goBack()
-
-    onClick?.()
+    if (onClick !== undefined && typeof onClick === 'function') onClick()
+    else history.goBack()
```
Comment (robinandeer) [inline src/client/components/BackButton/BackButton.tsx:28]: Wouldn't it be enough to do?

```suggestion
    if (onClick) return onClick()
    history.goBack()
```

**Review by robinandeer — COMMENTED** [review]
Great stuff!

Just a bit confused about defining variables before they are used - have a look at those comments.

**Review by guilhermespopolin — COMMENTED** [review]
Yeah, nice fix. Just agree with the "variable declaration" thing Robin has mentioned

**Review by robinandeer — APPROVED** [review]
Nice great stuff!

**Review by guilhermespopolin — APPROVED** [review]

**Pull request merged**